runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTranslationContext.java

```diff
@@ -68,13 +71,18 @@ public FlinkBatchTranslationContext(ExecutionEnvironment env, PipelineOptions op
     this.danglingDataSets = new HashMap<>();
   }
 
+  void init(Pipeline pipeline) {
+    pipeline.traverseTopologically(countingPipelineVisitor);
+    pipeline.traverseTopologically(lookupPipelineVisitor);
+  }
+
   // ------------------------------------------------------------------------
 
-  public Map<PValue, DataSet<?>> getDanglingDataSets() {
+  Map<PValue, DataSet<?>> getDanglingDataSets() {
```

Comment:
Can you revert these? This makes integration hard for external Runners.

Reply:
The whole class is marked as package private, so this change should not make any difference when it comes to 3rd code. Am I missing something?

Reply:
Didn't see that this was the case. It's still an unrelated change and it draws attention away from the actual issue.